src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,324 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Input,
+  OnDestroy,
+  Optional,
+  Output,
+} from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import {DateAdapter} from './date-adapter';
+import {Subscription} from 'rxjs';
+import {CdkDatepicker} from './datepicker';
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * An event used for datepicker input and change events. For consistency, we always use
+ * DatepickerInputEvent instead.
+ */
+export class DatepickerInputEvent<D> {
+  /** The new value for the target datepicker input. */
+  value: D | null;
+
+  constructor(
+    /** Reference to the datepicker input component that emitted the event. */
+    public target: CdkDatepickerInput<D>,
+      /** Reference to the native input element associated with the datepicker input. */
+      public targetElement: HTMLElement) {
+    this.value = this.target.value;
+  }
+}
+
+
+/** Directive used to connect an input to a CdkDatepicker. */
+@Directive({
+  selector: 'input[cdkDatepicker]',
+  providers: [
+    DATEPICKER_VALUE_ACCESSOR,
+    DATEPICKER_VALIDATORS,
```

Comment:
I see you removed a line here:
`{provide: MAT_INPUT_VALUE_ACCESSOR, useExisting: MatDatepickerInput},`

This is important for making the date input compatible with the mat input. We may need to move `MAT_INPUT_VALUE_ACCESSOR` into the cdk as well. Let's talk with @jelbourn about it

Reply:
We decided that this change will be located in material.